File: trunk/libs/python/jld/registry/exception.py
""" Registry Exception class
"""
__author__  = "Jean-Lou Dupont"
__version__ = "$Id$"

class RegistryException(Exception):
    """ An exception class for Registry
    """
    def __init__(self, value):
        self.value = value
    
    def __str__(self):
        return str(self.value)

File: libs/php/JLD/System/Exception.php
<?php
/**
 * Base exception class
 * Upon construction, posts an 'exception' message through the system.
 *
 * @author Jean-Lou Dupont
 * @package System
 * @subpackage Exception
 * @version @@package-version@@
 * @Id $Id$
 */
//<source lang=php>
require_once 'JLD/System/Logger.php';

class JLD_System_Exception extends Exception
{
	/**
	 * Constructor
	 */
	public function __construct( $message = null, $priority = LOG_INFO)
	{
		parent::__construct( $message, $priority );

		JLD_System_Logger::log( $priority, $message );		
	}
	
}//end class
//</source>